Extract.hs

```diff
@@ -0,0 +1,28 @@
+import System.IO
+import System.Environment
+import Data.List
+
+main = do
+    args <- getArgs
+    header <- return( args !! 0)
+    file <- return( args !! 1)
+    -- putStrLn ("Searching for '" ++ header ++ "' in " ++ file)
+    withFile file ReadMode (\handle -> do
+        contents <- hGetContents handle
+        putStr $ processText header contents)
+
+processText :: String -> String -> String
+processText tgt input =
+    let removed = dropWhile (\x -> not (matchesHeader tgt x)) $ lines input
+        lvl = getHeadingLevel (removed !! 0)
+        children = takeWhile (\x -> (matchesHeader tgt x) || getHeadingLevel x /= lvl) removed
+        result = unlines $ tail children
+    in  result
+
+matchesHeader :: String -> String -> Bool
+matchesHeader tgt line =
+  isInfixOf ("# " ++ tgt) line
```

Comment:
So this is complete personal preference, but `<>` is the general concat operator, and `++` is for only (I think) strings, so I tend to use `<>` everywhere.

```suggestion
  isInfixOf ("# " <> tgt) line
```